src/app/vault/bulk-delete.component.ts

```diff
@@ -18,17 +19,25 @@ import { I18nService } from 'jslib/abstractions/i18n.service';
 export class BulkDeleteComponent {
     @Input() cipherIds: string[] = [];
     @Input() permanent: boolean = false;
+    @Input() organization: Organization;
     @Output() onDeleted = new EventEmitter();
 
-    formPromise: Promise<any>;
-
     constructor(private analytics: Angulartics2, private cipherService: CipherService,
-        private toasterService: ToasterService, private i18nService: I18nService) { }
+        private toasterService: ToasterService, private i18nService: I18nService, private apiService: ApiService) { }
 
     async submit() {
-        this.formPromise = this.permanent ? this.cipherService.deleteManyWithServer(this.cipherIds) :
-            this.cipherService.softDeleteManyWithServer(this.cipherIds);
-        await this.formPromise;
+        if (!this.organization || !this.organization.isAdmin) {
+            this.permanent
+                ? await this.cipherService.deleteManyWithServer(this.cipherIds)
+                : await this.cipherService.softDeleteManyWithServer(this.cipherIds);
+        } else {
+            this.permanent
+                ? await this.apiService
+                    .deleteManyCiphersAdmin(new CipherBulkDeleteRequest(this.cipherIds, this.organization.id))
+                : await this.apiService
+                    .putDeleteManyCiphersAdmin(new CipherBulkDeleteRequest(this.cipherIds, this.organization.id));
```

Comment:
I think long-hand if/else would probably read better here

Reply:
Kinda done, I created some private methods that use long hand if/else and called those.